spinalcordtoolbox/math.py

```diff
@@ -402,6 +402,29 @@ def symmetrize(data, dim):
     return data_out
 
 
+def slicewise_mean(data, dim, exclude_zeros=False):
+    """
+    Symmetrize data along specified dimension. Does not count zeros in the mean.
+    :param data: numpy.array 3D data.
+    :param dim: dimension of array to symmetrize along.
+
+    :return data_out: slicewise averaged data
+    """
+    if exclude_zeros:
+        data[data == 0] = np.nan
+    data_out = np.zeros_like(data)
+    for slices in range(data.shape[dim]):
+        idx_to_slice = [slice(None)] * data.ndim
+        idx_to_slice[dim] = slices
```

Comment:
I'm sorry, I think I led you wrong here! I'm getting the following error when I run the test:

```python
    if np.isnan(data[idx_to_slice]).all():
# IndexError: only integers, slices (`:`), ellipsis (`...`), numpy.newaxis (`None`) 
# and integer or boolean arrays are valid indices
```

It looks like the syntax has changed, and numpy now requires a tuple of slice objects rather than a list. So we need to add:

```suggestion
        idx_to_slice = [slice(None)] * data.ndim
        idx_to_slice[dim] = slices
        idx_to_slice = tuple(idx_to_slice)  # requirement of numpy indexing
```